subsys/bluetooth/host/att.c

```diff
@@ -723,14 +723,19 @@ static struct net_buf *bt_att_chan_create_pdu(struct bt_att_chan *chan, uint8_t
 		return NULL;
 	}
 
-	switch (att_op_get_type(op)) {
-	case ATT_RESPONSE:
-	case ATT_CONFIRMATION:
-		/* Use a timeout only when responding/confirming */
-		timeout = BT_ATT_TIMEOUT;
-		break;
-	default:
-		timeout = K_FOREVER;
+	/* We can't block in the system workqueue */
+	if (k_current_get() == k_work_queue_thread_get(&k_sys_work_q)) {
+		timeout = K_NO_WAIT;
+	} else {
+		switch (att_op_get_type(op)) {
+		case ATT_RESPONSE:
+		case ATT_CONFIRMATION:
+			/* Use a timeout only when responding/confirming */
+			timeout = BT_ATT_TIMEOUT;
+			break;
+		default:
+			timeout = K_FOREVER;
+		}
```

Comment:
Would it make sense to only have the `if (k_current_get() == k_work_queue_thread_get(&k_sys_work_q)) {` in the `default` case? 

In the current version of GATT we should never have read and indication responses in the system workqueue but always in the BT RX thread

Reply:
That's ok with me

Reply:
Now updated